src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -564,42 +564,67 @@ protected void Accepting()
                 }
                 else if (policy.Item1 is Pass && policy.Item2 != null)
                 {
-                    var pass = policy.Item1 as Pass;
-                    if (pass.Uid == quarantine.Uid)
+                    var pass = (Pass) policy.Item1;
```

Comment:
This changes mirrors a bug report and subsequent fix Patrik Nordwall told us about: https://github.com/akka/akka/issues/20523

basically during `Quarantine` the `EndpointManager` would not properly account for the state of the quarantined UID and as a result, would leak `ReliableDeliverySupervisor` instances from previous incarnations of that association.

So now we explicitly update what the quarantined UID is inside the `EndpointRegistry` and check on `Quarantine` events to make sure that the policies are always being applied to the correct, current UID.

Additionally this fix cleans up the `GotUid` handler and will not correctly apply `Quarantined` policies if the node that joins has a UID that corresponds with the current `RefuseUid` value of the `Pass` policy.